tests/hardhat/utils/FixRevertError.ts

```diff
@@ -0,0 +1,27 @@
+import { assert } from "chai";
+import {
+  ContractFunctionExecutionError,
+  ContractFunctionRevertedError,
+  RawContractError,
+} from "viem";
+
+export const fixRevertError = (err: unknown) => {
+  assert(err instanceof ContractFunctionExecutionError);
+
+  const { abi, functionName, message, shortMessage } = err;
+
+  const { data } = (err.walk((err) => "data" in (err as Error)) ||
+    err.walk()) as RawContractError;
+
+  return new ContractFunctionRevertedError({
+    abi,
+    data: typeof data === "object" ? data.data : data,
+    functionName,
+    message: shortMessage ?? message,
+  });
+};
+
+export const fixRevertErrorIfNeeded = (err: unknown) => {
+  if (err instanceof ContractFunctionRevertedError) return err;
```

Comment:
When does this happen?

Reply:
Such behaviour is stated in the docs. It is done this way for more flexibility on potential `viem` updates.
Alternatively, we can only work with current situation and remove this function.

Reply:
> stated in the docs
Link?

Reply:
> on potential `viem` updates
Yeah, we'd want to disregard those that and fix our case. Here we aim to implement the most deterministic path possible as this is for the tests only.

Reply:
> Link?

I meant `viem` [docs](https://viem.sh/docs/contract/simulateContract#handling-custom-errors).

And I removed this code.

Reply:
I get it if you'd do it like this:

```ts
  if (err instanceof BaseError) {
    const revertError = err.walk(err => err instanceof ContractFunctionRevertedError)
    if (revertError instanceof ContractFunctionRevertedError) {
       return revertError;
    }
  }
  // ...
```